contracts/rewards/src/contract/execute.rs

```diff
@@ -42,23 +43,28 @@ where
     /// block height and the epoch duration. If the epoch duration is updated, we store the epoch
     /// in which the update occurs as the last checkpoint
     fn current_epoch(&self, cur_block_height: u64) -> Result<Epoch, ContractError> {
+        assert!(
+            cur_block_height <= i64::MAX as u64,
```

Comment:
Why do all the calculations in `i64` when everything is `u64`?

Reply:
I also have this question

Reply:
I tried to use this to have a nicely aligned match statement, but I changed it now